transcendence-app/src/chat/chat.gateway.ts

```diff
@@ -30,10 +32,45 @@ export class ChatGateway {
   constructor(
     private chatService: ChatService,
     private chatroomMemberService: ChatroomMemberService,
+    private userService: UserService,
   ) {}
 
+  @SubscribeMessage('chatMessage')
+  async handleChatMessage(
+    @MessageBody()
+    createChatMessageDto: CreateChatMessageDto,
+    @ConnectedSocket() client: Socket,
+  ) {
+    const { userId: recipientId, content } = createChatMessageDto;
+
+    const senderId = client.request.user.id;
+
+    const block = await this.userService.getBlockRelation(
+      recipientId,
+      senderId,
+    );
+
+    if (block?.isUserBlockedByMe || block?.amIBlockedByUser) {
+      throw new WsException('There is an existing blockage. Forbidden.');
+    }
+
+    const message = await this.chatService.saveOneToOneChatMessage({
+      senderId,
+      recipientId,
+      content,
+    });
+
+    if (message) {
+      this.server.to(recipientId).emit('chatMessage', { ...message });
```

Comment:
suggestion: If I'm not mistaken, we also need to emit the message to the sender.

Reply:
You mean if we the user has multiple tabs/apps wathever open right? I'll add it 👍

Reply:
why do we need to spread the message if we're not modifiying it at all?

Reply:
> suggestion: If I'm not mistaken, we also need to emit the message to the sender.

If you don't mind I will update this in next PR, I've made other backend changes

> why do we need to spread the message if we're not modifiying it at all?

Because we only draw the messages we receive by ws. I tought at first that we directly draw, in webapp, a message once we send it. But we send it, save it, and receive it. I think this is good so we only see persisted messages